python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx

```diff
@@ -108,6 +118,176 @@ const MainNavBar = () => {
   const rootRouteId = mainNavPageHierarchy[0]?.id;
   const { metricsContextLoaded, grafanaHost } = useContext(GlobalContext);
 
+  const [timezone, setTimezone] = useState<string>(() => {
+    const currentTimezone =
+      localStorage.getItem("timezone") ||
+      Intl.DateTimeFormat().resolvedOptions().timeZone;
+    formatTimeZone(currentTimezone);
+    return currentTimezone;
+  });
+
+  useEffect(() => {
+    localStorage.setItem("timezone", timezone);
+  }, [timezone]);
+
+  const handleTimezoneChange = (event: SelectChangeEvent<string>) => {
+    setTimezone(event.target.value);
+    window.location.reload();
+  };
+
+  const timezones = [
```

Comment:
where did this list come from? Does this cover most common time zones?

Reply:
This list was exported from GitHub's Timezone selection options.
![image](https://github.com/user-attachments/assets/4fee6387-12a7-4b39-8b0c-26f4bdfcf783)